File: questao_2/node/palindrome.test.js
import { before, describe, it } from "node:test";
import assert from "node:assert";
import isPalindrome from "./palindrome.js";

describe("palindrome", () => {
  let sut = null;
  before(() => {
    sut = isPalindrome;
  });

  it("should return ? if length < 3", () => {
    const useCases = ["a", "ab", "aba"];
    useCases.forEach((data) => {
      assert.equal(sut(data), "?");
    });
  });

  it("should return N if palindrome is not valid", () => {
    const useCases = ["ABAA", "ASDF", "ABBAA"];
    useCases.forEach((data) => {
      assert.equal(sut(data), "N");
    });
  });

  it("should return S if palindrome is valid", () => {
    const useCases = ["ABBA", "aabaa", "cabac"];
    useCases.forEach((data) => {
      assert.equal(sut(data), "S");
    });
  });
});

File: questao_2/java/src/PalindromeTest.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import org.junit.runner.Result;
import org.junit.runner.JUnitCore;
import org.junit.runner.notification.Failure;

public class PalindromeTest {
  @Test
  public void testIsPalindrome() {
    assertEquals('S', Palindrome.isPalindrome("ABBA"));
  }

}
